# Intervalos de confianza


In [16]:
import numpy as np
import matplotlib.pyplot as plt
import scipy.stats as st

## 1. Intervalo de confianza para la media de una población normal:


Sea $X_1, \dots, X_n$ una m.a $X_i\sim\mathcal{N}(\mu,\sigma^2)$ emtpnces

- $\bar{X} \sim \mathcal{N}(\mu,\sigma^2/n)$
- $Z = \frac{\bar{X}-\mu}{\sigma/n}\sim \mathcal{N}(0,1)$


### - $\sigma$ conocida


Se tiene que el intervalo del $(1-\alpha)x100\%$ de confianza para $\mu$ es:

$$\bar{X} \pm z_{\alpha/2}\frac{\sigma}{\sqrt(n)}$$


**Ejemplo 1** Para tratar de estimar la media de consumo por cliente en un restaurante, se reunieron datos de una muestra de tamaño 49 clientes durante tres semanas.

**a)** Supongamos que la desviación estándar de la población es de $ 2.5 $. ¿Cuál es el error estándar de la media?

**b)** Con un nivel de confianza del 95\%, ¿cuál es el margen de error?

**c)** Si la media de la muestra es de $ 22.60 $, ¿cuál es el intervalo de confianza del 95\% para la media de la población?


In [17]:
n = 49 #clientes
sigma = 2.5 #desviación estándar de la población
media_muestral = 22.60
confianza = 0.95

# a) SE
error_est = sigma / np.sqrt(n)
print(f"Error estándar: {error_est}")

# b) 1 - (1 - confianza) / 2
z = st.norm.ppf(1 - (1 - confianza) / 2)
margen_error = z * error_est
print(f"El margen de error es: {margen_error}")

# c) Intervalo de confianza
lim_inf = media_muestral - margen_error
lim_sup = media_muestral + margen_error
print(f"Intervalo de confianza del {confianza*100}%: ({lim_inf}, {lim_sup})")

Error estándar: 0.35714285714285715
El margen de error es: 0.6999871373357336
Intervalo de confianza del 95.0%: (21.90001286266427, 23.299987137335734)


Tenemos un 95\% de confianza de que el verdadero consumo poblacional en el restaurante se encuentra entre $\$21.90$ y $\$23.30$


**Ejemplo 2** Supongamos que se toma una muestra aleatoria de 100 personas para estimar la media del peso de una población, y se obtiene que la media muestral es de 70 kg con una desviación estándar que es conocida de 10 kg. Para un nivel de confianza del 95\%, calcular el intervalo de confianza.


In [18]:
# Datos
n = 100
sigma = 10
media_muestral = 70
confianza = 0.95
error_est = sigma / np.sqrt(n)

# Utilizaremos la función 
# scypy.stats.norm.interval para calcular el intervalo de confianza

intervalo = st.norm.interval(confianza, loc=media_muestral, scale=error_est)
print(f"Intervalo de confianza del {confianza*100}%: {intervalo[0]:.2f}, {intervalo[1]:.2f}")

Intervalo de confianza del 95.0%: 68.04, 71.96


Por lo tanto, con un 95% de confianza, estimamos que el peso medio verdadero de la población se encuentra entre $68.04$ y $71.96$


> Se podría practicar modificando el nivel de confianza al $90\%$ o $99\%$ para ver como cambia el intervalo.


### - $\sigma$ desconocida


Se tiene que el intervalo del $(1-\alpha)x 100\%$ de confianza para $\mu$ es:

$$ \bar{X} \pm t\_{\alpha/2}\frac{\sigma}{\sqrt n}$$


**Ejemplo 3** Supongamos que tenemos los siguientes datos de una muestra elatoria de 10 observaciones:

$$ 45,55, 67, 45, 68, 79, 98, 87, 84, 82$$

Clacular un intervalo de confianza para la media


In [19]:
# Datos
datos = np.array([45, 55, 67, 45, 68, 79, 98, 87, 84, 82])

n = len(datos)
confianza = 0.95
gl = n - 1

media_muestral = np.mean(datos)
error_est = st.sem(datos) # error estándar de la media muestral

#Utilizaremos la función
#scipy.stats.t.interval para calcular el intervalo de confianza
intervalo = st.t.interval(confianza, gl, loc=media_muestral, scale=error_est)
print(f"Media muestral: {media_muestral:.2f}")
print(f"Intervalo de confianza del {confianza*100}%: {intervalo[0]:.2f}, {intervalo[1]:.2f}")

Media muestral: 71.00
Intervalo de confianza del 95.0%: 58.00, 84.00


**Ejemplo 4** Los articulos de cerámica utilizados sobre velas eléctricas se rompen con diferentes presiones.
Supongamos que los datos provienen de una distribución normal. La resistencia a la ruptura fue medida con una muestra de 100 articulos y el promedio fue de 1750 con una desviación estandar de 315.8.

**a)** Estimar con un nivel de confianza del $90\%$ a la media poblacional de la presión de la ruptura.

**b)** Estimar con un nivel de confianza del $90\%$ a la varianza poblacional.

$$
\left(\frac{(n-1)s^2}{\chi^2_{1-\alpha/2}}, \frac{(n-1)s^2}{\chi^2_{\alpha/2}}\right)
$$


In [20]:
# Datos
n = 100
confianza = 0.90
gl = n - 1
media_muestral = 1750
desviacion_estandar = 315.8

# a) Utilizamos z ya que n es mayor a 30
error_est = desviacion_estandar / np.sqrt(n)
inttervalo = st.norm.interval(confianza, loc=media_muestral, scale=error_est)
print(f"Intervalo de confianza del {confianza*100}% para la media: {inttervalo[0]:.2f}, {inttervalo[1]:.2f}")

# b) IC para la varianza Chi-cuadrada
alpha = 1 - confianza
s2 = desviacion_estandar ** 2 # varianza muestral

# Forma 1. calculando los cuantiles de la chi-cuadrada
chi2_inf = st.chi2.ppf(alpha / 2, gl)
chi2_sup = st.chi2.ppf(1 - alpha / 2, gl)
int_var = ((gl*s2) / chi2_sup,gl*s2/chi2_inf)
print(f"Intervalo de confianza del {confianza*100}% para la varianza: ({int_var[0]:.2f},{int_var[1]:.2f})")

# Forma 2.
chi2_inf,chi2_sup = st.chi2.interval(confianza,gl)
int_var2 = ((gl*s2) / chi2_sup,gl*s2/chi2_inf)
print(f"Intervalo de confianza del {confianza*100}% para la varianza: ({int_var2[0]:.2f},{int_var2[1]:.2f})")


Intervalo de confianza del 90.0% para la media: 1698.06, 1801.94
Intervalo de confianza del 90.0% para la varianza: (80123.49,128146.72)
Intervalo de confianza del 90.0% para la varianza: (80123.49,128146.72)


## Tarea


**Ejercicio 2** Los datos $1.2,\ 2.1,\ 2.3,\ 1.9,\ 3.0,\ 1.5,\ 2.8,\ 2.4,\ 1.7,\ 2.6,\ 2.0,\ 1.8,\ 3.1,\ 2.2,\ 2.5$ se toman a partir de una distribución normal $N(\mu,\sigma^2)$ con $\mu$ desconocida.

**a)** Encuentra un intervalo de confianza al nivel de $90\%$ para $\mu$, dado que $\sigma =2$.

**b)** Encuentra un intervalo de confianza al nivel de $90\%$ para $\mu$.

**c)** Encuentra un intervalo de confianza al nivel de $90\%$ para $\sigma^2$.


In [ ]:
# Datos
datos = np.array([1.2, 2.1, 2.3, 1.9, 3.0, 1.5, 2.8, 2.4, 1.7, 2.6, 2.0, 1.8, 3.1, 2.2, 2.5])
n = len(datos)
confianza = 0.90
gl = n - 1
media_muestral = np.mean(datos)

# a) sigma conocida (sigma = 2) con distribución normal
sigma = 2
error_est = sigma / np.sqrt(n)
intervalo_a = st.norm.interval(confianza, loc=media_muestral, scale=error_est)
print(f"a) Intervalo de confianza del {confianza*100}% para mu (sigma = {sigma} conocida):({intervalo_a[0]:.2f}, {intervalo_a[1]:.2f})")

# b) sigma desconocida con distribución t de Student
error_est_t = st.sem(datos)
intervalo_b = st.t.interval(confianza, gl, loc=media_muestral, scale=error_est_t)
print(f"b) Intervalo de confianza del {confianza*100}% para mu (sigma desconocida): ({intervalo_b[0]:.2f}, {intervalo_b[1]:.2f})")

# c) IC para la varianza con distribución chi-cuadrada
s2 = np.var(datos)
chi2_inf, chi2_sup = st.chi2.interval(confianza, gl)
int_var = ((gl * s2) / chi2_sup, (gl * s2) / chi2_inf)
print(f"c) Intervalo de confianza del {confianza*100}% para sigma^2: ({int_var[0]:.2f}, {int_var[1]:.2f})")

a) Intervalo de confianza del 90.0% para mu (sigma = 2 conocida):(1.36, 3.06)
b) Intervalo de confianza del 90.0% para mu (sigma desconocida): (1.96, 2.45)
c) Intervalo de confianza del 90.0% para sigma^2: (0.16, 0.59)
